panics when trying to create scalar type []byte

### Expected Behaviour
I expect to be able to create my own implementation of scalar Bytes as a go type `[]byte`

### Actual Behavior
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x18 pc=0x12d6b93]

goroutine 1 [running]:
github.com/99designs/gqlgen/codegen/config.(*TypeReference).UniquenessKey(0xc0004bcea0, 0xc0004bcea0, 0xc0005647a0)
        /github.com/99designs/gqlgen@v0.8.1/codegen/config/binder.go:290 +0x83
github.com/99designs/gqlgen/codegen.(*builder).buildTypes(0xc000289cc0, 0xc000330000, 0x0, 0x0)
        /github.com/99designs/gqlgen@v0.8.1/codegen/type.go:12 +0x86
github.com/99designs/gqlgen/codegen.BuildData(0xc00016c180, 0x140e9e0, 0x166d1d8, 0x0)
        /github.com/99designs/gqlgen@v0.8.1/codegen/data.go:119 +0x884
github.com/99designs/gqlgen/api.Generate(0xc00016c180, 0x0, 0x0, 0x0, 0x0, 0x0)
       /github.com/99designs/gqlgen@v0.8.1/api/generate.go:37 +0x228
main.main()
        /main.go:23 +0x82
exit status 2

### Minimal graphql.schema and models to reproduce
schema.graphqls
```graphql
scalar Bytes

type Query {
    test(in: Bytes): Boolean
}
```
gqlgen.yml
```yaml
schema:
  - ./schema.graphqls
exec:
  filename: ./gql/exec.go
models:
  Bytes:
    model: github.com/me/myrepo/types.Bytes
```
types.go
```go

func MarshalBytes(b []byte) graphql.Marshaler {
	return graphql.WriterFunc(func(w io.Writer) {
		_, _ = fmt.Fprintf(w, "%q", string(b))
	})
}

func UnmarshalBytes(v interface{}) ([]byte, error) {
	switch v := v.(type) {
	case string:
		return []byte(v), nil
	case *string:
		return []byte(*v), nil
	case []byte:
		return v, nil
	default:
		return nil, fmt.Errorf("%T is not []byte", v)
	}
}
```

--- a/codegen/config/binder.go
+++ b/codegen/config/binder.go
@@ -193,9 +193,9 @@ func (ref *TypeReference) Elem() *TypeReference {
 		}
 	}
 
-	if s, isSlice := ref.GO.(*types.Slice); isSlice {
+	if ref.IsSlice() {
 		return &TypeReference{
-			GO:          s.Elem(),
+			GO:          ref.GO.(*types.Slice).Elem(),
 			GQL:         ref.GQL.Elem,
 			CastType:    ref.CastType,
 			Definition:  ref.Definition,
@@ -221,7 +221,7 @@ func (t *TypeReference) IsNilable() bool {
 
 func (t *TypeReference) IsSlice() bool {
 	_, isSlice := t.GO.(*types.Slice)
-	return isSlice
+	return t.GQL.Elem != nil && isSlice
 }
 
 func (t *TypeReference) IsNamed() bool {

--- a/codegen/testserver/bytes.go
+++ b/codegen/testserver/bytes.go
@@ -1,0 +1,27 @@
+package testserver
+
+import (
+	"fmt"
+	"io"
+
+	"github.com/99designs/gqlgen/graphql"
+)
+
+func MarshalBytes(b []byte) graphql.Marshaler {
+	return graphql.WriterFunc(func(w io.Writer) {
+		_, _ = fmt.Fprintf(w, "%q", string(b))
+	})
+}
+
+func UnmarshalBytes(v interface{}) ([]byte, error) {
+	switch v := v.(type) {
+	case string:
+		return []byte(v), nil
+	case *string:
+		return []byte(*v), nil
+	case []byte:
+		return v, nil
+	default:
+		return nil, fmt.Errorf("%T is not []byte", v)
+	}
+}

--- a/codegen/testserver/generated.go
+++ b/codegen/testserver/generated.go
@@ -172,6 +172,7 @@ type ComplexityRoot struct {
 		Overlapping            func(childComplexity int) int
 		Panics                 func(childComplexity int) int
 		Recursive              func(childComplexity int, input *RecursiveInputSlice) int
+		ScalarSlice            func(childComplexity int) int
 		ShapeUnion             func(childComplexity int) int
 		Shapes                 func(childComplexity int) int
 		Slices                 func(childComplexity int) int
@@ -270,6 +271,7 @@ type QueryResolver interface {
 	Panics(ctx context.Context) (*Panics, error)
 	DefaultScalar(ctx context.Context, arg string) (string, error)
 	Slices(ctx context.Context) (*Slices, error)
+	ScalarSlice(ctx context.Context) ([]byte, error)
 	Fallback(ctx context.Context, arg FallbackToStringEncoding) (FallbackToStringEncoding, error)
 	OptionalUnion(ctx context.Context) (TestUnion, error)
 	ValidType(ctx context.Context) (*ValidType, error)
@@ -777,6 +779,13 @@ func (e *executableSchema) Complexity(typeName, field string, childComplexity in
 		}
 
 		return e.complexity.Query.Recursive(childComplexity, args["input"].(*RecursiveInputSlice)), true
+
+	case "Query.ScalarSlice":
+		if e.complexity.Query.ScalarSlice == nil {
+			break
+		}
+
+		return e.complexity.Query.ScalarSlice(childComplexity), true
 
 	case "Query.ShapeUnion":
 		if e.complexity.Query.ShapeUnion == nil {
@@ -1312,6 +1321,7 @@ scalar Time
 `},
 	&ast.Source{Name: "slices.graphql", Input: `extend type Query {
     slices: Slices
+    scalarSlice: Bytes!
 }
 
 type Slices {
@@ -1320,6 +1330,8 @@ type Slices {
   test3: [String]!
   test4: [String!]!
 }
+
+scalar Bytes
 `},
 	&ast.Source{Name: "typefallback.graphql", Input: `extend type Query {
     fallback(arg: FallbackToStringEncoding!): FallbackToStringEncoding!
@@ -3763,6 +3775,33 @@ func (ec *executionContext) _Query_slices(ctx context.Context, field graphql.Col
 	rctx.Result = res
 	ctx = ec.Tracer.StartFieldChildExecution(ctx)
 	return ec.marshalOSlices2ᚖgithubᚗcomᚋ99designsᚋgqlgenᚋcodegenᚋtestserverᚐSlices(ctx, field.Selections, res)
+}
+
+func (ec *executionContext) _Query_scalarSlice(ctx context.Context, field graphql.CollectedField) graphql.Marshaler {
+	ctx = ec.Tracer.StartFieldExecution(ctx, field)
+	defer func() { ec.Tracer.EndFieldExecution(ctx) }()
+	rctx := &graphql.ResolverContext{
+		Object:   "Query",
+		Field:    field,
+		Args:     nil,
+		IsMethod: true,
+	}
+	ctx = graphql.WithResolverContext(ctx, rctx)
+	ctx = ec.Tracer.StartFieldResolverExecution(ctx, rctx)
+	resTmp := ec.FieldMiddleware(ctx, nil, func(rctx context.Context) (interface{}, error) {
+		ctx = rctx // use context from middleware stack in children
+		return ec.resolvers.Query().ScalarSlice(rctx)
+	})
+	if resTmp == nil {
+		if !ec.HasError(rctx) {
+			ec.Errorf(ctx, "must not be null")
+		}
+		return graphql.Null
+	}
+	res := resTmp.([]byte)
+	rctx.Result = res
+	ctx = ec.Tracer.StartFieldChildExecution(ctx)
+	return ec.marshalNBytes2ᚕbyte(ctx, field.Selections, res)
 }
 
 func (ec *executionContext) _Query_fallback(ctx context.Context, field graphql.CollectedField) graphql.Marshaler {
@@ -6620,6 +6659,20 @@ func (ec *executionContext) _Query(ctx context.Context, sel ast.SelectionSet) gr
 				res = ec._Query_slices(ctx, field)
 				return res
 			})
+		case "scalarSlice":
+			field := field
+			out.Concurrently(i, func() (res graphql.Marshaler) {
+				defer func() {
+					if r := recover(); r != nil {
+						ec.Error(ctx, ec.Recover(ctx, r))
+					}
+				}()
+				res = ec._Query_scalarSlice(ctx, field)
+				if res == graphql.Null {
+					invalid = true
+				}
+				return res
+			})
 		case "fallback":
 			field := field
 			out.Concurrently(i, func() (res graphql.Marshaler) {
@@ -7206,6 +7259,14 @@ func (ec *executionContext) unmarshalNBoolean2bool(ctx context.Context, v interf
 
 func (ec *executionContext) marshalNBoolean2bool(ctx context.Context, sel ast.SelectionSet, v bool) graphql.Marshaler {
 	return graphql.MarshalBoolean(v)
+}
+
+func (ec *executionContext) unmarshalNBytes2ᚕbyte(ctx context.Context, v interface{}) ([]byte, error) {
+	return UnmarshalBytes(v)
+}
+
+func (ec *executionContext) marshalNBytes2ᚕbyte(ctx context.Context, sel ast.SelectionSet, v []byte) graphql.Marshaler {
+	return MarshalBytes(v)
 }
 
 func (ec *executionContext) unmarshalNDefaultScalarImplementation2string(ctx context.Context, v interface{}) (string, error) {

--- a/codegen/testserver/gqlgen.yml
+++ b/codegen/testserver/gqlgen.yml
@@ -68,3 +68,5 @@ models:
       oldFoo: { fieldName: foo, resolver: true }
   FallbackToStringEncoding:
     model: "github.com/99designs/gqlgen/codegen/testserver.FallbackToStringEncoding"
+  Bytes:
+    model: "github.com/99designs/gqlgen/codegen/testserver.Bytes"

--- a/codegen/testserver/resolver.go
+++ b/codegen/testserver/resolver.go
@@ -140,6 +140,9 @@ func (r *queryResolver) DefaultScalar(ctx context.Context, arg string) (string, 
 func (r *queryResolver) Slices(ctx context.Context) (*Slices, error) {
 	panic("not implemented")
 }
+func (r *queryResolver) ScalarSlice(ctx context.Context) ([]byte, error) {
+	panic("not implemented")
+}
 func (r *queryResolver) Fallback(ctx context.Context, arg FallbackToStringEncoding) (FallbackToStringEncoding, error) {
 	panic("not implemented")
 }

--- a/codegen/testserver/slices.graphql
+++ b/codegen/testserver/slices.graphql
@@ -1,5 +1,6 @@
 extend type Query {
     slices: Slices
+    scalarSlice: Bytes!
 }
 
 type Slices {
@@ -8,3 +9,5 @@ type Slices {
   test3: [String]!
   test4: [String!]!
 }
+
+scalar Bytes

--- a/codegen/testserver/slices_test.go
+++ b/codegen/testserver/slices_test.go
@@ -30,4 +30,16 @@ func TestSlices(t *testing.T) {
 		require.NotNil(t, resp.Slices.Test3)
 		require.NotNil(t, resp.Slices.Test4)
 	})
+
+	t.Run("custom scalars to slices work", func(t *testing.T) {
+		resolvers.QueryResolver.ScalarSlice = func(ctx context.Context) ([]byte, error) {
+			return []byte("testing"), nil
+		}
+
+		var resp struct {
+			ScalarSlice string
+		}
+		c.MustPost(`query { scalarSlice }`, &resp)
+		require.Equal(t, "testing", resp.ScalarSlice)
+	})
 }

--- a/codegen/testserver/stub.go
+++ b/codegen/testserver/stub.go
@@ -50,6 +50,7 @@ type Stub struct {
 		Panics                 func(ctx context.Context) (*Panics, error)
 		DefaultScalar          func(ctx context.Context, arg string) (string, error)
 		Slices                 func(ctx context.Context) (*Slices, error)
+		ScalarSlice            func(ctx context.Context) ([]byte, error)
 		Fallback               func(ctx context.Context, arg FallbackToStringEncoding) (FallbackToStringEncoding, error)
 		OptionalUnion          func(ctx context.Context) (TestUnion, error)
 		ValidType              func(ctx context.Context) (*ValidType, error)
@@ -191,6 +192,9 @@ func (r *stubQuery) DefaultScalar(ctx context.Context, arg string) (string, erro
 }
 func (r *stubQuery) Slices(ctx context.Context) (*Slices, error) {
 	return r.QueryResolver.Slices(ctx)
+}
+func (r *stubQuery) ScalarSlice(ctx context.Context) ([]byte, error) {
+	return r.QueryResolver.ScalarSlice(ctx)
 }
 func (r *stubQuery) Fallback(ctx context.Context, arg FallbackToStringEncoding) (FallbackToStringEncoding, error) {
 	return r.QueryResolver.Fallback(ctx, arg)

